app/src/main/java/com/github/studydistractor/sdp/procrastinationActivity/FireBaseProcrastinationActivityService.kt

```diff
@@ -37,7 +37,17 @@ class FireBaseProcrastinationActivityService @Inject constructor(): Procrastinat
             })
     }
 
-    override fun postProcastinationActivities(activity: ProcrastinationActivity) {
-        databaseRef.child(activity.name!!).setValue(activity)
+    override fun postProcastinationActivities(
+        activity: ProcrastinationActivity,
+        onSuccess: () -> Unit,
+        onFailure: () -> Unit
+    ) {
+        databaseRef.child(activity.name!!).setValue(activity).addOnCompleteListener { task ->

```

Comment:
Don't forget to detach listeners for this data after completion of the task. https://firebase.google.com/docs/database/android/read-and-write

Reply:
Done!